lib/codecs/src/decoding/format/protobuf.rs

```diff
@@ -0,0 +1,353 @@
+use std::collections::BTreeMap;
+use std::fs;
+use std::path::PathBuf;
+
+use bytes::Bytes;
+use chrono::Utc;
+use ordered_float::NotNan;
+use prost_reflect::{DescriptorPool, DynamicMessage, MessageDescriptor, ReflectMessage};
+use smallvec::{smallvec, SmallVec};
+use vector_config::configurable_component;
+use vector_core::event::LogEvent;
+use vector_core::{
+    config::{log_schema, DataType, LogNamespace},
+    event::Event,
+    schema,
+};
+use vrl::value::Kind;
+
+use super::Deserializer;
+
+/// Config used to build a `ProtobufDeserializer`.
+#[configurable_component]
+#[derive(Debug, Clone, Default)]
+pub struct ProtobufDeserializerConfig {
+    /// Path to desc file
+    desc_file: PathBuf,
+
+    /// message type. e.g package.message
+    message_type: String,
+}
+
+impl ProtobufDeserializerConfig {
+    /// Build the `ProtobufDeserializer` from this configuration.
+    pub fn build(&self) -> ProtobufDeserializer {
+        // TODO return a Result instead.
+        ProtobufDeserializer::try_from(self).unwrap()
```

Comment:
I know we said we'll follow up on this TODO but I think we should have resolved it before this was merged since otherwise we have to keep track of it to resolve before the next release. To that end I created this issue to track it: https://github.com/vectordotdev/vector/issues/18110 and put it in a new 0.32.0 milestone to make sure we resolve it prior to the next release.